web-client/src/pages/admin/AdminSockDetailsPage.tsx

```diff
@@ -0,0 +1,244 @@
+      refetch();
+      setEditSockOpen(false);
+      toast.success("Sock updated successfully!");
+    } catch (error) {
+      console.error("Error updating sock:", error);
+      toast.error("Failed to update sock details.");
+    }
+  };
+
+  const handleAddVariant = async (newVariant: SockVariant) => {
+    try {
+      const updatedVariants: SockVariant[] = [
+        ...currentVariants,
+        {
+          ...newVariant,
+          size: newVariant.size as SockVariant["size"],
+        },
+      ];
+      setCurrentVariants(updatedVariants);
+
+      await updateSockMutation.mutateAsync({
+        id: numericSockId,
+        sock: {
+          name: sock!.name,
+          description: sock!.description,
+          previewImageUrl: sock!.previewImageUrl,
+        },
+        variants: updatedVariants,
+      });
+
+      refetch();
+      setAddVariantOpen(false);
+      toast.success("Variant added successfully!");
+    } catch (error) {
+      console.error("Error adding variant:", error);
+      toast.error("Failed to add variant.");
+    }
+  };
+
+  const handleEditVariant = async (updatedVariant: SockVariant) => {
+    try {
+      const updatedVariants: SockVariant[] = currentVariants.map((variant) =>
+        variant.id === updatedVariant.id ? updatedVariant : variant
+      );
+      setCurrentVariants(updatedVariants);
+
+      await updateSockMutation.mutateAsync({
+        id: numericSockId,
+        sock: {
+          name: sock!.name,
+          description: sock!.description,
+          previewImageUrl: sock!.previewImageUrl,
+        },
+        variants: updatedVariants,
+      });
+
+      refetch();
+      setEditVariant(null);
+      toast.success("Variant updated successfully!");
+    } catch (error) {
+      console.error("Error updating variant:", error);
+      toast.error("Failed to update variant.");
+    }
+  };
+
+  if (isLoading) return <div>Loading...</div>;
+  if (isError || !sock) return <div>Error fetching sock details.</div>;
+
+  const availableSizes = ["S", "M", "LG", "XL"].filter(
+    (size) => !currentVariants.some((variant) => variant.size === size)
+  );
+
+  return (
+    <div className="max-w-7xl mx-auto p-8">
+      {/* Header Section */}
+      <div className="flex items-center justify-between mb-8">
+        <h1 className="text-3xl font-bold">Item Details: {sock.name}</h1>
+        <Button variant="default" onClick={() => setEditSockOpen(true)}>
+          <Edit className="mr-2" size={16} /> Edit Item
+        </Button>
+      </div>
+
+      {/* Image and Basic Information */}
+      <div className="grid grid-cols-1 md:grid-cols-2 gap-8">
+        {/* Item Image */}
+        <div className="flex justify-center items-center">
+          <img
+            src={sock.previewImageUrl || NO_IMAGE_PLACEHOLDER}
+            alt={sock.name}
+            className="w-full max-w-[500px] h-auto object-contain border rounded-lg shadow-md md:max-w-[700px]"
```

Comment:
Think the image is a bit too tall. Make it be full width and the make the height 400px. And no shadow is fine actually